### ISSUE TITLE
Questions on marginal distance constraints in the code

### ISSUE THREAD

**Issue by Cheeun** [zysxmu/IntraQ#6]
Thanks for sharing your code. It was very helpful in understanding your interesting work.
I have some questions about your code.

For generating the data for cifar100, it seems that the **marginal distance constraints** (loss_cosineDistance and loss_cosineDistance_upper in distill_data.py) **are both 0** during the data generation process.
Is this a code error or am I missing something?

Also, could you explain why you used **1-CosineSimilarilty** instead of CosineSimilarity in your code? 

Thanks in advance!

**Comment by zysxmu** [comment]
Why are they 0? At the beginning of the training process? Or are they always 0?

We use the cosine distance, which is 1-cosine similarilty.

**Comment by Cheeun** [comment]
Thanks for such a quick answer!
According to my results for reproducing, they are always 0.
For the loss_cosineDistance_lower, it seems that cosine distance always becomes larger than the lower cosine margin, thus the loss is always clipped to zero. Similar for loss_cosineDistance_upper, as well.

**Comment by zysxmu** [comment]
What would happen if set loss_cosineDistance_lower to a large value or set loss_cosineDistance_upper to a small value?

In my experiment, this loss would be non-zero in the data generation process of CIFAR100. At least in these iterations of the last stage.

**Comment by Cheeun** [comment]
Thanks for your quick reply.
I tried changing (cosineMargin, cosineMargin_upper) to (0.2, 0.4), which was originally (0.02, 1.0). 
Now, the loss_cosineDistance is a non-zero value, but still loss_cosineDistance_upper is mostly zero.
With these generated samples, the quantized accuracy is quite similar to that of the samples generated when the loss_cosineDistances are both always zero.
So this makes me a bit curious about the **effect of loss_cosineDistances (and the effect of these cosine margin hyperparameters)**.
It seems that loss is effective for Imagenet (according to the table 5 in the paper), but does this also work for cifar100?

**Comment by zysxmu** [comment]
Well, according to my experiments long ago, it indeed exhibits some effects.

But we should take the recent experiments as the standard.

So, yes, it may didn't work very well for CIFAR100.
